### PR TITLE
[Compiler-v2] Fix issue 12121

### LINKED ISSUE #12121: [Bug][Compiler-v2] Inconsistent error message while compiling a script
# 🐛 Bug

When compiling the following code:

```
script {
fun main() {
    if (true) {
        y = 5;
    } else {
        y = 0;
    };
    assert!(y == 5, 42);
}
}
```

V2 will generate the following error:

```
Diagnostics:
error: undeclared `y`
  ┌─ tests/checking/typing/v1-commands/no_let_outside_if.move:4:9
  │
4 │         y = 5;
  │         ^

error: undeclared `y`
  ┌─ tests/checking/typing/v1-commands/no_let_outside_if.move:6:9
  │
6 │         y = 0;
  │         ^

error: undeclared `<SELF>_0::y`
  ┌─ tests/checking/typing/v1-commands/no_let_outside_if.move:8:13
  │
8 │     assert!(y == 5, 42);
  │             ^

```

Need to investigate why the last message has '<SELF>_0...` while previous ones don't.

### PR DESCRIPTION
**Pull request by rahxephon89**
### Description

This PR closes #12121 by printing the name without qualifier.

### Test Plan

Error messages are changed.

### CODE REVIEW

File: third_party/move/move-model/src/builder/exp_builder.rs [hunk 1]
```diff
@@ -2371,10 +2371,12 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
...
         }
-
         self.error(
             loc,
-            &format!("undeclared `{}`", global_var_sym.display(self.env())),
+            &format!(
+                "undeclared `{}`",
+                global_var_sym.symbol.display(self.symbol_pool())
+            ),
         );
```
[the remark this answers is not in the document]
Comment (vineethk) [reply third_party/move/move-model/src/builder/exp_builder.rs:2380]: Also, there seemed to be at least one more such instance of qualified symbol being displayed in `translate_macro_call` (currently at line 3599) - but I was not able to trigger a `<Self_0>::name` style output easily for that part of code.

Perhaps take a brief look there too?

**Review by vineethk — APPROVED** [review]
Code change suggested, LGTM otherwise.

File: third_party/move/move-model/src/builder/exp_builder.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (brmataptos) [reply third_party/move/move-model/src/builder/exp_builder.rs:2380]: This way lies a lot of work.  I see a dozen uses of `QualifiedSymbol::display` and 2 uses of `QualifiedSymbol::display_full`.  Maybe check them all?

**Review by brmataptos — APPROVED** [review]
Looks good for solving this particular problem.  Could add an issue to check all uses of `QualifiedSymbol::display`, or check them now if you like.

File: third_party/move/move-model/src/builder/exp_builder.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (rahxephon89) [reply third_party/move/move-model/src/builder/exp_builder.rs:2380]: I skimmed all of them and it seems that they are used intentionally. Will file a issue for it.

File: third_party/move/move-model/src/builder/exp_builder.rs [hunk 2]
```diff
@@ -2471,10 +2471,9 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
                 vec![global_access.into_exp()],
             );
         }
-
         self.error(
             loc,
-            &format!("undeclared `{}`", global_var_sym.display(self.env())),
+            &format!("undeclared `{}`", global_var_sym.display_simple(self.env())),
```
Comment (wrwg) [inline third_party/move/move-model/src/builder/exp_builder.rs:2476]: You can't do it like this. What if the user actually provided a qualified name (see line 2392)?

File: third_party/move/move-model/src/builder/exp_builder.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (wrwg) [reply third_party/move/move-model/src/builder/exp_builder.rs]: If a name is qualified then indeed it should be printed as qualified...

File: third_party/move/move-model/src/builder/exp_builder.rs [hunk 3]
```diff
@@ -3737,7 +3740,7 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
                     loc,
                     &format!(
                         "`{}` is a function and not a macro",
-                        qsym.display(self.env())
+                        qsym.display_simple(self.env())
```
Comment (rahxephon89) [inline third_party/move/move-model/src/builder/exp_builder.rs:3743]: Check the error message and it makes more sense to print the function name only.

File: third_party/move/move-model/src/builder/exp_builder.rs (hunk 2 above)
Comment (rahxephon89) [reply third_party/move/move-model/src/builder/exp_builder.rs]: Added a check in this commit: https://github.com/aptos-labs/aptos-core/pull/12316/commits/aaacb5eade784c36a3f133e9f16a4b5b9b913992

**Review by wrwg — APPROVED** [review]

**Pull request merged**